63 and 64: tinyint/smallint and time/data types

avalanche123 opened the pull request:
This is a work in progress.
Unfortunately, I think I've uncovered a bug in metadata parsing in cpp driver.
The attached test case always causes "Invalid value" exception.
After adding a print statement in `execute_request.hpp`, `get_type()`, I see that the type of value is considered a custom type `org.apache.cassandra.db.marshal.ShortType` instead of `smallint`.

@mpenick let me know if you need help reproducing or if it turns out that my implementation is at fault

Commit:
update doc stubs

Commit:
fix negative limit

Commit:
[PHP-63] add smallint

Commit:
fix error messages in unit tests

Commit:
only surround strings with quotes in error messages

mpenick commented:
what cpp-driver function is returning the error? Is it binding a value to a prepared statement? I've tested this using C\* 2.2 and 3.X and haven't been able to reproduce it yet.

mpenick commented:
Here's the example used: https://gist.github.com/mpenick/c70e743cbed922d9e5d3c587874c82da

Commit:
[PHP-63] switch to protocol v4 as the default

Commit:
[PHP-63] smallint integration test

Commit:
[PHP-63] add tinyint

Commit:
[PHP-65] add support for Date and Time types

Commit:
fix docblock

Commit:
remove unused variable

Commit:
Fixed min/max for tinyint

Commit:
Added Time::now()

Commit:
Added Date::fromDateTime() and Time::fromDateTime()

Commit:
Fixing Windows, C89, and PHP v5.5 build issues

NOTE: This does not address link issues for dynamic library due to
      internal use of date/lib; statically linked executable is
      possible.

mikefero commented on ext/src/Cassandra/Date.c:
```diff
@@ -97,14 +100,51 @@ PHP_METHOD(Date, toDateTime)
+  }
+  timelib_update_ts(datetime_obj->time, NULL);
+
+  timestamp = timelib_date_to_int(dateobj->time, &error);
```
**`timelib_update_ts`** and **`timelib_date_to_int`** is an internal function and is not exported in the PHP dynamic library. Linking the extension statically into the executable is possible; however a DLL is not.  Since these functions are used in both `Time.c` and `Date.c` a utils implementation would be best.

Commit:
Fixing linux build

Commit:
Correcting PHP validation for timelib_long

Commit:
Unit tests and fixes for date and time

jippi commented:
Any alive-ness of this one? 

Also, are there a way to insert `date` fields without this PR ?

mpenick commented:
@jippi sorry, been working on the cpp-driver. We're working on getting this out ASAP.

jippi commented:
Thanks - cost a lot of grief last week :(

Commit:
Various fixes for date/time/tinyint types

Commit:
test: Fix constant and assert test for cpp-driver v2.4.2

Commit:
fix: Correcting invalid type returned for tinyint

Commit:
test: Adding new types to collection tests and simplifying scalar tests

Commit:
test: Fixing broken tests

mpenick commented:
Merged at 0d2159ee8dfe2feaa50c5ccf164f032a9b990874
